### ISSUE TITLE
DUK_USE_ROM_GLOBAL_INHERIT has incorrect provideThis for global env

### ISSUE THREAD

**Issue by svaarala** [svaarala/duktape#1340]
When using ROM objects and using the `DUK_USE_ROM_GLOBAL_INHERIT` option, the global object environment record has "provideThis" incorrectly set. Correct behavior is:

```
duk> function foo() { 'use strict'; print(Object.prototype.toString.call(this)); }; foo();
[object Undefined]
```

The buggy behavior is:

```
duk> function foo() { 'use strict'; print(Object.prototype.toString.call(this)); }; foo();
[object global]
```

This mostly matters for global function bindings declared as "use strict". Instead of getting the correct `undefined` this binding (which gets coerced to the global object for non-strict functions) the functions will get the global object as the this binding.

**Comment by svaarala** [comment]
This was fixed by #1310 but needs a backport (separate fix) to 2.0.2 and 1.7.0.

**Comment by svaarala** [comment]
Now tracked in the maintenance milestones.
